src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -814,6 +814,7 @@ def make_cell_lines(self, entry):
     def make_donor(self, donor):
         return {
             "id": donor["biomaterial_id"],
+            "donorCount": donor["donor_count"],
```

Comment:
The IT fails when hitting the service's health endpoint which in turns hits the repository endpoints. Hitting those manually after enabling AZUL_DEBUG yields.

```
$ http 'https://service.sandbox.dev.explore.data.humancellatlas.org/repository/bundles?size=1'
HTTP/1.1 500 Internal Server Error

Traceback (most recent call last):
  File "/var/task/chalice/app.py", line 726, in _get_view_function_response
    response = view_function(**function_args)
  File "/var/task/app.py", line 580, in get_bundle_data
    return repository_search('bundles', bundle_uuid)
  File "/var/task/app.py", line 435, in repository_search
    return service.get_data(entity_type, pagination, filters, item_id, file_url)
  File "/var/task/azul/service/repository.py", line 86, in get_data
    return self._get_items(entity_type, pagination, filters, file_url_func)
  File "/var/task/azul/service/repository.py", line 60, in _get_items
    response = self._get_data(entity_type, pagination, filters, file_url_func)
  File "/var/task/azul/service/repository.py", line 39, in _get_data
    entity_type=entity_type)
  File "/var/task/azul/service/responseobjects/elastic_request_builder.py", line 526, in transform_request
    final_response = FileSearchResponse(hits, paging, facets, entity_type)
  File "/var/task/azul/service/responseobjects/hca_response_v5.py", line 990, in __init__
    KeywordSearchResponse.__init__(self, hits, entity_type)
  File "/var/task/azul/service/responseobjects/hca_response_v5.py", line 898, in __init__
    self.map_entries(x) for x in hits], 'pagination': None}
  File "/var/task/azul/service/responseobjects/hca_response_v5.py", line 898, in <listcomp>
    self.map_entries(x) for x in hits], 'pagination': None}
  File "/var/task/azul/service/responseobjects/hca_response_v5.py", line 882, in map_entries
    donorOrganisms=self.make_donors(entry),
  File "/var/task/azul/service/responseobjects/hca_response_v5.py", line 827, in make_donors
    return [self.make_donor(donor) for donor in entry["contents"]["donors"]]
  File "/var/task/azul/service/responseobjects/hca_response_v5.py", line 827, in <listcomp>
    return [self.make_donor(donor) for donor in entry["contents"]["donors"]]
  File "/var/task/azul/service/responseobjects/hca_response_v5.py", line 817, in make_donor
    "donorCount": donor["donor_count"],
KeyError: 'donor_count'
```

So while it is a good default approach to use `[]` for keys that you expect to be present, this creates a problem during the integration test which hits the service endpoint *before* a reindexing can be triggered in Gitlab manually. I think we should use `.get` as for the other document fields.

Reply:
Ah, so in other words, this function needs to process donors that were assembled by older versions of the code that don't have the other changes I made. Got it.

Reply:
Yes, there is a little bit of a chicken-egg problem. We don't want to reindex until we've vetted the code but in this case the vetting fails because we haven't reindex yet. I'm currently unsure as to how we'll solve this.

Reply:
I would like the service code to be strict when it comes to consuming the index, but at the moment that's not really possible.